integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java

```diff
@@ -63,16 +66,18 @@
 import org.apache.spark.sql.streaming.Trigger;
 import org.apache.spark.sql.types.StructType;
 import org.awaitility.Awaitility;
-import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
 import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
 import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
 import org.testcontainers.utility.DockerImageName;
 
 @Slf4j
 @Tag("integration-test")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
What is the reason for doing this? I am wondering if the same behaviour can be achieved with using `@Testcontainers` annotation as used for example here within `io.openlineage.spark.agent.ColumnLineageIntegrationTest`

Reply:
I ve used this only to keep handler object instance per class not per tests to not loose events. Spark still starts session with the same config for open lineage, I can use trick with a file but I think its a little tricky.